lib/query_hook.rb

```diff
@@ -6,10 +6,39 @@ def tempfile_extension
   end
 
   def compile_file_content(r)
-    "#{r.extra}\n#{r.content}\nprint('=> ' + (#{r.query}).inspect)"
+    <<ruby
+    #{r.extra}
+    #{r.content}
+    #{compile_cookie(r.cookie)}
+    #{compile_query(r.query)}
+ruby
   end
 
   def command_line(filename)
     "ruby #{filename}"
   end
-end
\ No newline at end of file
+
+  def compile_query(query)
+    if query.start_with? 'def '
+      "#{query}\nputs \"<function>\""
```

Comment:
I think that we should print 'nil' here, since in ruby, defining a method returns always `nil`